# Nombre: Nixon Reynaldo Rojas Guevara
# Codigo 2231944

# Parte I
# Instruccion 1

In [68]:
import numpy as np

In [69]:
np.random.seed(0)
datos = np.random.randint(118, 190, 50)


In [70]:
a = datos[datos > 118]
a

array([162, 165, 182, 185, 185, 127, 139, 154, 188, 130, 176, 183, 157,
       164, 155, 143, 127, 138, 187, 165, 182, 167, 147, 137, 137, 132,
       157, 150, 183, 127, 175, 150, 149, 141, 153, 173, 146, 152, 154,
       171, 123, 156, 135, 122, 160, 176, 149, 119])

In [88]:
import pandas as pd
x = pd.DataFrame(a)
normalizar = lambda x: (x - a.mean()) / a.std()
resultado = normalizar(x)
print(resultado)

           0
0   0.362469
1   0.515535
2   1.382911
3   1.535977
4   1.535977
5  -1.423303
6  -0.811038
7  -0.045707
8   1.689043
9  -1.270237
10  1.076778
11  1.433933
12  0.107359
13  0.464513
14  0.005315
15 -0.606950
16 -1.423303
17 -0.862060
18  1.638021
19  0.515535
20  1.382911
21  0.617580
22 -0.402862
23 -0.913082
24 -0.913082
25 -1.168193
26  0.107359
27 -0.249796
28  1.433933
29 -1.423303
30  1.025756
31 -0.249796
32 -0.300818
33 -0.708994
34 -0.096729
35  0.923712
36 -0.453884
37 -0.147751
38 -0.045707
39  0.821668
40 -1.627391
41  0.056337
42 -1.015127
43 -1.678413
44  0.260425
45  1.076778
46 -0.300818
47 -1.831480


In [117]:
import pandas as pd
df1 = pd.DataFrame(a)
df1

,0
0,162
1,165
2,182
3,185
4,185
5,127
6,139
7,154
8,188
9,130


In [118]:
df1.describe()

,0
count,48.000000
mean,154.895833
std,19.806768
min,119.000000
25%,138.750000
50%,154.000000
75%,171.500000
max,188.000000


.describe(), lo que hace es contar, sacar la media la desviacion estandar, el minimo,el maximo, tambien dice que el 25% de los datos estan en 138, el 50% esta en 154 y el 75% esta en 171

In [76]:
import pandas as pd
df = pd.DataFrame(datos)
df

,0
0,162
1,165
2,182
3,185
4,185
5,127
6,139
7,154
8,188
9,130


In [98]:
from google.colab import files
import csv
resultados = [datos]

with open("resultados.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(resultados)

files.download("resultados.csv")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Instruccion 2

In [81]:
import pandas as pd
df = pd.DataFrame({"nombre": ["Laura", "Nixon", "Karen", "Majo", "Angie"],
                   "edad" : [23, 20, 29, 21, 20],
                   "nota" : [4.5, 3.8, 4.2, 4.0, 4.1],
                   "deporte": ["voleybol", "futbol", "baloncesto", "", "futbol"]})
df

,nombre,edad,nota,deporte
0,Laura,23,4.5,voleybol
1,Nixon,20,3.8,futbol
2,Karen,29,4.2,baloncesto
3,Majo,21,4.0,
4,Angie,20,4.1,futbol


In [100]:
df.head(1)

,nombre,edad,nota,deporte
0,Laura,23,4.5,voleybol


In [83]:
df["deporte"]

,deporte
0,voleybol
1,futbol
2,baloncesto
3,
4,futbol


In [85]:
df.loc[df["edad"] >= 23]

,nombre,edad,nota,deporte
0,Laura,23,4.5,voleybol
2,Karen,29,4.2,baloncesto


In [110]:
df.sum()

,0
nombre,LauraNixonKarenMajoAngie
edad,113
nota,20.6
deporte,voleybolfutbolbaloncestofutbol


In [103]:
#calcular deporte y edad
grupo1 = df.groupby("deporte")["edad"].mean()
grupo1

,edad
deporte,
,21.0
baloncesto,29.0
futbol,20.0
voleybol,23.0


In [104]:
#nombre y nota
grupo2 = df.groupby("nombre")["nota"].mean()
grupo2

,nota
nombre,
Angie,4.1
Karen,4.2
Laura,4.5
Majo,4.0
Nixon,3.8


In [106]:
#edad y deporte
grupo3 = df.groupby("edad")["deporte"].count()
grupo3

,deporte
edad,
20,2
21,1
23,1
29,1


In [109]:
from google.colab import files
import csv
datos_procesados = [df]

with open("datos_procesados.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos_procesados)

files.download("datos_procesados.csv")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Parte III

In [123]:
import numpy as np
import pandas as pd


Prueba_A = np.random.normal(loc=7.2, scale=0.4, size=52)
Prueba_B = np.random.gamma(shape=306, scale=0.0229, size=52)


In [124]:
df2 = pd.DataFrame({
    "A": Prueba_A,
    "B": Prueba_B
    })

In [125]:
df2.describe()

,A,B
count,52.000000,52.000000
mean,7.212722,6.934927
std,0.443235,0.335127
min,6.137752,6.093275
25%,6.898840,6.726830
50%,7.176074,6.965138
75%,7.601262,7.170391
max,8.278490,7.532346


Descripcion:
segun el count hay 52 elemento, seguen el mean la media de A es de 7.2 y la media de B es de 6.9, la desviacion estandar de A es de  0.4 y de la B es 0.3, el minimo de A es de 6.13, y el minimo de B es de 6.09, en el cuantil 25% de A es de 6.8 y de B es de 6.7, el cuantil de A es de 7.17 y el de B es de 6.9, el cuantil de 75% es de 7.60 y el de B es 7.17 y el maximo de A es de 8.27 y el de B es de 7.5.

In [126]:
from scipy import stats

In [131]:
from scipy.stats import shapiro
shapiro([Prueba_A])

ShapiroResult(statistic=np.float64(0.9879782910257657), pvalue=np.float64(0.8755126020884614))

In [132]:
from scipy.stats import shapiro
shapiro([Prueba_B])

ShapiroResult(statistic=np.float64(0.9807582419309185), pvalue=np.float64(0.558592702097092))

# En ambas pruebas  si se cumple el supuesto de normalidad con una confianza del 95%

In [133]:
stats.ttest_rel(Prueba_A, Prueba_B)


TtestResult(statistic=np.float64(3.492207540727073), pvalue=np.float64(0.0009986690122736773), df=np.int64(51))

# si se rechaza que el puntaje medio en ambas encuestas es igual con una confianza del 95%

In [134]:
stats.kendalltau(Prueba_A, Prueba_B)



SignificanceResult(statistic=np.float64(-0.016591251885369532), pvalue=np.float64(0.8621769404880528))

# No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados

In [147]:
t_stats, p_val = stats.spearmanr(df2["A"], df2["B"])
print("Estadístico t:", t_stats)
print("Valor p:", p_val)

Estadístico t: -0.024331938871339533
Valor p: 0.8640506881081852


# No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados

# Punto 8, 9 y 10 no los entendi :(